src/sage/matroids/matroid.pyx

```diff
@@ -3032,20 +3178,81 @@ cdef class Matroid(SageObject):
                     next_level.extend(Ht)
         return B
 
+    def no_broken_circuits_sets_iterator(self, ordering=None):
+        r"""
+        Return an iterator over no broken circuits (NBC) sets of ``self``.
+
+        An NBC set is a subset `A` of the groundset under some total
+        ordering `<` such that `A` contains no broken circuit.
+
+        INPUT:
+
+        - ``ordering`` -- a total ordering of the groundset given as a list
+
+        EXAMPLES::
+
+            sage: M = Matroid(circuits=[[1,2,3], [3,4,5], [1,2,4,5]])
+            sage: SimplicialComplex(list(M.no_broken_circuits_sets_iterator()))
+            Simplicial complex with vertex set (1, 2, 3, 4, 5)
+             and facets {(1, 2, 4), (1, 2, 5), (1, 3, 4), (1, 3, 5)}
+            sage: SimplicialComplex(list(M.no_broken_circuits_sets_iterator([5,4,3,2,1])))
+            Simplicial complex with vertex set (1, 2, 3, 4, 5)
+             and facets {(1, 3, 5), (1, 4, 5), (2, 3, 5), (2, 4, 5)}
+
+        ::
+
+            sage: M = Matroid(circuits=[[1,2,3], [1,4,5], [2,3,4,5]])
+            sage: SimplicialComplex(list(M.no_broken_circuits_sets_iterator([5,4,3,2,1])))
+            Simplicial complex with vertex set (1, 2, 3, 4, 5)
+             and facets {(1, 3, 5), (2, 3, 5), (2, 4, 5), (3, 4, 5)}
+        """
+        if not self.loops():
```

Comment:
This is a slight change from what we current have for a matroid with loops., where it returns the empty set. I am not sure offhand what behavior is correct. In either case, we should document it. I would also do this (possibly with a `yield emptyset()`) to avoid having a big indentation level too:
```suggestion
        if self.loops():
            return
        [rest of method]
```

Reply:
For a matroid with loops the correct behavior is no sets at all. I am planning to do some more work on the broken circuit methods in another PR.